lib/src/solver/package_lister.dart

```diff
@@ -36,6 +36,13 @@ class PackageLister {
   /// solve isn't a `pub get`.
   final PackageId _locked;
 
+  // The version of this package that, if retracted, is still allowed in the
+  // current version solve.
+  //
+  // We don't allow a retracted version during solving unless it is already
+  // present in pubspec.lock or pinned in dependency_overrides.
```

Comment:
```suggestion
  // present in `pubspec.lock` or pinned in `dependency_overrides`.
```

Reply:
Should this just be a `Version`, or not a `PackageId`? maybe it doesn't matter...

Reply:
Please document that this is `null` if no exceptions are to be made.